pkg/operatorsource/poller.go

```diff
@@ -45,28 +47,48 @@ type Poller interface {
 type poller struct {
 	helper    PollHelper
 	datastore datastore.Writer
+	sender    PackageUpdateNotificationSender
 }
 
 func (p *poller) Poll() {
 	sources := p.datastore.GetAllOperatorSources()
 
+	aggregator := datastore.NewPackageUpdateAggregator()
+
 	for _, source := range sources {
-		if err := p.pollSource(source); err != nil {
+		result, err := p.helper.HasUpdate(source)
+		if err != nil {
+			log.Errorf("[sync] error checking for updates [%s] - %v", source.Name, err)
+			continue
+		}
+
+		if !result.RegistryHasUpdate {
+			continue
+		}
+
+		log.Infof("operator source[%s] has updates: %s", source.Name, result)
+		aggregator.Add(result)
+
+		if err := p.trigger(source, result); err != nil {
 			log.Errorf("%v", err)
 		}
 	}
-}
 
-func (p *poller) pollSource(source *datastore.OperatorSourceKey) error {
-	updated, err := p.helper.HasUpdate(source)
-	if err != nil {
-		return fmt.Errorf("[sync] error checking for updates [%s] - %v", source.Name, err)
+	// We have a list of operator(s) that have either been removed or have new
+	// version(s). We should kick off CatalogSourceConfig reconciliation.
+	if !aggregator.IsUpdatedOrRemoved() {
+		return
 	}
 
-	if !updated {
-		return nil
-	}
+	// TODO: This is a stop gap measure. We should not need this any longer when
+	// CatalogSourceConfig has the version stored.
+	<-time.After(time.Minute * 10)
```

Comment:
Can we make this 10 minute value a `const` somewhere else?

Reply:
moved to `main.go` so that all the duration consts are in one place.